mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/value/ValueDeclarer.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.component.value;
+
+import static java.util.Collections.unmodifiableList;
+import static org.mule.runtime.internal.util.collection.UnmodifiableMap.unmodifiableMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ValueDeclarer {
+
+  public static SimpleValue ofSimpleValue(Object value) {
+    return new ImmutableSimpleValue(value);
+  }
+
+  public static ArrayValueBuilder ofArrayValue() {
+    return new ArrayValueBuilder();
+  }
+
+  public static MapValueBuilder ofMapValue() {
+    return new MapValueBuilder();
+  }
+
+  public static ObjectValueBuilder ofObjectValue() {
+    return new ObjectValueBuilder();
+  }
```

Comment:
wouldn't this always be ObjectValues from the client code standpoint?

Reply:
I did not understand the question, are you making reference to the MapValue?